### PR TITLE
added broadcast again

### PR DESCRIPTION
**Pull request by fokkonaut**

### CODE REVIEW

File: src/game/client/components/broadcast.cpp [hunk 1]
```diff
@@ -19,6 +19,16 @@ CBroadcast::CBroadcast()
 	OnReset();
 }
 
+void CBroadcast::OnMessage(int MsgType, void *pRawMsg)
+{
+	if (MsgType == NETMSGTYPE_SV_BROADCAST)
```
Comment (ChillerDragon) [inline src/game/client/components/broadcast.cpp]:
```suggestion
	if(MsgType == NETMSGTYPE_SV_BROADCAST)
```

File: src/game/client/components/broadcast.cpp [hunk 2]
```diff
@@ -28,6 +38,27 @@ void CBroadcast::DoBroadcast(const char *pText)
 	TextRender()->TextEx(&Cursor, m_aBroadcastText, -1);
 	m_BroadcastRenderOffset = 150*Graphics()->ScreenAspect()-Cursor.m_X/2;
 	m_BroadcastTime = time_get()+time_freq()*10;
+
+	char aBuf[1024];
+	int i, ii;
+	for (i = 0, ii = 0; i < str_length(m_aBroadcastText); i++)
```
Comment (ChillerDragon) [inline src/game/client/components/broadcast.cpp]:
```suggestion
	for(i = 0, ii = 0; i < str_length(m_aBroadcastText); i++)
```

File: src/game/client/components/broadcast.cpp [hunk 3]
```diff
@@ -28,6 +38,27 @@ void CBroadcast::DoBroadcast(const char *pText)
...
+	{
+		if (m_aBroadcastText[i] == '\n')
```
Comment (ChillerDragon) [inline src/game/client/components/broadcast.cpp]:
```suggestion
		if(m_aBroadcastText[i] == '\n')
```

File: src/game/client/components/broadcast.cpp [hunk 4]
```diff
@@ -28,6 +38,27 @@ void CBroadcast::DoBroadcast(const char *pText)
...
+		{
+			aBuf[ii] = '\0';
+			ii = 0;
+			if (aBuf[0])
```
Comment (ChillerDragon) [inline src/game/client/components/broadcast.cpp]:
```suggestion
			if(aBuf[0])
```

File: src/game/client/components/broadcast.cpp [hunk 5]
```diff
@@ -28,6 +38,27 @@ void CBroadcast::DoBroadcast(const char *pText)
...
+				m_pClient->Console()->Print(IConsole::OUTPUT_LEVEL_STANDARD, "broadcast", aBuf, true);
+		}
+		else
+		{
+			aBuf[ii] = m_aBroadcastText[i];
+			ii++;
+		}
+	}
+	aBuf[ii] = '\0';
+	if (aBuf[0])
```
Comment (ChillerDragon) [inline src/game/client/components/broadcast.cpp]:
```suggestion
	if(aBuf[0])
```

File: src/game/server/gamecontext.cpp [hunk 6]
```diff
@@ -1083,6 +1090,31 @@ void CGameContext::ConRestart(IConsole::IResult *pResult, void *pUserData)
...
 
+void CGameContext::ConBroadcast(IConsole::IResult *pResult, void *pUserData)
+{
+	CGameContext *pSelf = (CGameContext *)pUserData;
+
+	char aBuf[1024];
+	str_copy(aBuf, pResult->GetString(0), sizeof(aBuf));
+
+	int i, j;
+	for (i = 0, j = 0; aBuf[i]; i++, j++)
```
Comment (ChillerDragon) [inline src/game/server/gamecontext.cpp]:
```suggestion
	for(i = 0, j = 0; aBuf[i]; i++, j++)
```

File: src/game/server/gamecontext.cpp [hunk 7]
```diff
@@ -1083,6 +1090,31 @@ void CGameContext::ConRestart(IConsole::IResult *pResult, void *pUserData)
...
+	{
+		if (aBuf[i] == '\\' && aBuf[i + 1] == 'n')
```
Comment (ChillerDragon) [inline src/game/server/gamecontext.cpp]:
```suggestion
		if(aBuf[i] == '\\' && aBuf[i + 1] == 'n')
```

File: src/game/server/gamecontext.cpp [hunk 8]
```diff
@@ -1083,6 +1090,31 @@ void CGameContext::ConRestart(IConsole::IResult *pResult, void *pUserData)
...
+		{
+			aBuf[j] = '\n';
+			i++;
+		}
+		else if (i != j)
```
Comment (ChillerDragon) [inline src/game/server/gamecontext.cpp]:
```suggestion
		else if(i != j)
```

**Review by ChillerDragon — CHANGES_REQUESTED** [review]

**Comment by oy** [comment]
Won't be added that way. The DoBroadcast rendering area is reserved for client-sided notification now and nothing gonna interfere with that.

**Comment by ChillerDragon** [comment]
@oy what do you suggest doing then?

**Comment by oy** [comment]
@ChillerDragon not sure. Needs some thinking.

**Comment by nheir** [comment]
Maybe duplicating broadcast to have "GameMessage" and "Broadcast" where "GameMessage" is client build message as now, and Broadcast is the server build message.
Either with a exclusion (ir GameMessage forbidding Broadcast) or in slightly different zone so the messages can't overlap and may be distinguished (different font, straight vs italic...)

**Comment by Dune-jr** [comment]
`DoBroadcast` is actually used little (2 times), a lot of the text printing is done manually with adjusted parameters.

>Either with a exclusion (ir GameMessage forbidding Broadcast) or in slightly different zone so the messages can't overlap and may be distinguished (different font, straight vs italic...)

I agree. Maybe server broadcast could be some **smaller** text (in between the timer and the current warmup?). It should probably be made clear, by one way or another, that it is a custom text sent by the server and not a client message.

I think they still have their places in the game though - after all the `say` server messages are still around for custom uses.

**Comment by oy** [comment]
Different area probably best.

**Comment by ChillerDragon** [comment]
Imo the old broadcast area was pretty good. What about removing the broadcast if other messages appear?

**Comment by Dune-jr** [comment]
Can we look for compromises instead of resisting to just about any change? @ChillerDragon

**Comment by oy** [comment]
Putting it at the old spot means it will never show up f.e. in warmups, pauses and most of the time in survival.

**Pull request closed without merging**